lib/icinga/notification.cpp

```diff
@@ -428,9 +428,14 @@ void Notification::BeginExecuteNotification(NotificationType type, const CheckRe
 			continue;
 		}
 
+		// Verify if the 'Problem' filter is configured at both the User and Notification object levels.
+		bool foundProblemFilter = NotificationProblem & user->GetTypeFilter() && NotificationProblem & GetTypeFilter();
+
 		/* on recovery, check if user was notified before */
 		if (type == NotificationRecovery) {
```

Comment:
I don't see a point in doing this in advance. What if type doesn't match?

Reply:
> I don't see a point in doing this in advance.

The point is to deduplicate the logic, ensuring that changes to one part will automatically be reflected in the other, preventing any oversight.

> What if type doesn't match?

And what issue do you see in that case? It's just a simple boolean flag performing bitwise checks, so the type matching shouldn't be relevant.